airbyte-webapp/src/packages/cloud/views/layout/MainView/MainView.tsx

```diff
@@ -35,8 +36,9 @@ const MainView: React.FC<React.PropsWithChildren<unknown>> = (props) => {
   const alertToShow = showCreditsBanner ? "credits" : cloudWorkspace.trialExpiryTimestamp ? "trial" : undefined;
   // exp-speedy-connection
   const { isExperimentVariant } = useExperimentSpeedyConnection();
+  const { hasCorporateEmail } = useAuthService();
   const isTrial = Boolean(cloudWorkspace.trialExpiryTimestamp);
-  const showExperimentBanner = isExperimentVariant && isTrial;
+  const showExperimentBanner = isExperimentVariant && isTrial && hasCorporateEmail();
```

Comment:
There is currently one more issue with the existing implementation. In `DefaultView` we're navigating to the new connection creation page if that experiment is active, ignoring whether we're actually in a trial of the experiment banner. That e.g. causes us to also navigate to the new connection creation page when loging in and already having connecitons. That still needs to be fixed before activating this experiment, to make sure we only do the redirect if we're actually also showing the banner, i.e. we should likely move all this code here into the hook itself and use that as well in `DefaultView`.

Reply:
Discussed this offline. I was referencing a wrong experiment here. We're solving this as part of a separate PR.

Reply:
Solved here: #19382